tests/e2e-pw/specs/merchant/merchant-orders-partial-refund.spec.ts

```diff
@@ -0,0 +1,219 @@
+			{
+				lineItems: [
+					[ product1, 1 ],
+					[ product2, 2 ],
+					[ product3, 1 ],
+				],
+				refundInputs: [
+					{ refundQty: 1, refundAmount: 18 },
+					{ refundQty: 1, refundAmount: 55 },
+				],
+			},
+		],
+	];
+
+	let orderIds: string[];
+	let orderTotal: string;
+	let wasMulticurrencyEnabled = false;
+	let merchantPage: Page, shopperPage: Page;
+
+	const orderProducts = async ( dataTableIndex: number ) => {
+		const lineItems = dataTable[ dataTableIndex ][ 1 ].lineItems;
+		await goToShop( shopperPage );
+		await setupProductCheckout( shopperPage, lineItems );
+		await fillCardDetails( shopperPage );
+		await placeOrder( shopperPage );
+		await expect(
+			shopperPage.getByRole( 'heading', { name: 'Order received' } )
+		).toBeVisible();
+
+		const orderIdField = shopperPage.locator(
+			'.woocommerce-order-overview__order.order > strong'
+		);
+		const orderId = await orderIdField.innerText();
+
+		return orderId;
+	};
+
+	test.beforeAll( async ( { browser } ) => {
+		merchantPage = ( await getMerchant( browser ) ).merchantPage;
+		wasMulticurrencyEnabled = await activateMulticurrency( merchantPage );
+		await restoreCurrencies( merchantPage );
+		shopperPage = ( await getShopper( browser ) ).shopperPage;
+		const firstOrderId = await orderProducts( 0 );
+		const secondOrderId = await orderProducts( 1 );
+		orderIds = [ firstOrderId, secondOrderId ];
+	} );
+
+	test.afterAll( async () => {
+		if ( ! wasMulticurrencyEnabled ) {
+			await deactivateMulticurrency( merchantPage );
+		}
+	} );
+
+	dataTable.forEach( ( [ title, { refundInputs } ], i ) => {
+		test( title, async ( { browser } ) => {
+			merchantPage = ( await getMerchant( browser ) ).merchantPage;
+			await goToOrder( merchantPage, orderIds[ i ] );
+
+			const orderTotalField = merchantPage
+				.getByRole( 'row', { name: 'Order Total: $' } )
+				.locator( 'bdi' );
+
+			// Calculate order total, refund total, and net payment.
+			orderTotal = await orderTotalField.innerText();
+			const orderTotalNumber = parseFloat( orderTotal.substring( 1 ) );
+			const refundTotal = refundInputs
+				.map( ( { refundAmount } ) => refundAmount )
+				.reduce( ( acc, cur ) => acc + cur );
+			const refundTotalString = refundTotal.toFixed( 2 );
+			const netPayment = ( orderTotalNumber - refundTotal ).toFixed( 2 );
+
+			await merchantPage
+				.getByRole( 'button', { name: 'Refund' } )
+				.click();
+
+			// Fill in the refund inputs.
+			for ( let j = 0; j < refundInputs.length; j++ ) {
+				const { refundQty, refundAmount } = refundInputs[ j ];
+				await merchantPage
+					.locator( '.refund_order_item_qty' )
+					.nth( j )
+					.fill( refundQty.toString() );
+
+				await merchantPage
+					.locator( '.refund_line_total' )
+					.nth( j )
+					.clear();
+
+				await merchantPage
+					.locator( '.refund_line_total' )
+					.nth( j )
+					.fill( refundAmount.toString() );
+
+				await merchantPage.keyboard.press( 'Tab' );
```

Comment:
Not an issue but out of curiousity, why do we need this Tab press if we're filling with locators? To trigger the calculations?

Reply:
I found that I needed to move the focus from the last refund amount input otherwise the refund total wouldn't update to include the final refund amount.